lib/src/widgets/adw/avatar.dart

```diff
@@ -0,0 +1,206 @@
+        Color(0xFFd68400),
+      ],
+    ),
+    AdwAvatarColors.orange: AdwAvatarColor(
+      foregroundColor: Color(0xFFffe5c5),
+      backgroundGradient: [
+        Color(0xFFffa95a),
+        Color(0xFFed5b00),
+      ],
+    ),
+    AdwAvatarColors.raspberry: AdwAvatarColor(
+      foregroundColor: Color(0xFFf8d2ce),
+      backgroundGradient: [
+        Color(0xFFf78773),
+        Color(0xFFe62d42),
+      ],
+    ),
+    AdwAvatarColors.magenta: AdwAvatarColor(
+      foregroundColor: Color(0xFFfac7de),
+      backgroundGradient: [
+        Color(0xFFe973ab),
+        Color(0xFFe33b6a),
+      ],
+    ),
+    AdwAvatarColors.purple: AdwAvatarColor(
+      foregroundColor: Color(0xFFe7c2e8),
+      backgroundGradient: [
+        Color(0xFFcb78d4),
+        Color(0xFF9945b5),
+      ],
+    ),
+    AdwAvatarColors.violet: AdwAvatarColor(
+      foregroundColor: Color(0xFFd5d2f5),
+      backgroundGradient: [
+        Color(0xFF9e91e8),
+        Color(0xFF7a59ca),
+      ],
+    ),
+    AdwAvatarColors.beige: AdwAvatarColor(
+      foregroundColor: Color(0xFFf2eade),
+      backgroundGradient: [
+        Color(0xFFe3cf9c),
+        Color(0xFFb08952),
+      ],
+    ),
+    AdwAvatarColors.brown: AdwAvatarColor(
+      foregroundColor: Color(0xFFe5d6ca),
+      backgroundGradient: [
+        Color(0xFFbe916d),
+        Color(0xFF785336),
+      ],
+    ),
+    AdwAvatarColors.gray: AdwAvatarColor(
+      foregroundColor: Color(0xFFd8d7d3),
+      backgroundGradient: [
+        Color(0xFFc0bfbc),
+        Color(0xFF6e6d71),
+      ],
+    ),
+  };
+
+  /// Retunrs a set of values depending on the `color` variable.
+  ///
+  /// If the variable is `null`, it'll then return a random set of colors.
+  static AdwAvatarColor getColor(AdwAvatarColors? color) {
+    if (color != null) return _palette[color]!;
+
+    final randomColor =
+        AdwAvatarColors.values[Random().nextInt(AdwAvatarColors.values.length)];
+
+    return _palette[randomColor]!;
+  }
+}
+
+/// `AdwAvatar` is a widget that shows a round avatar. Usually, an icon or some
+/// letters. Automatic theme will be applied in those cases to make them look
+/// cohesive inside the avatar.
+class AdwAvatar extends StatelessWidget {
+  const AdwAvatar({
+    Key? key,
+    required this.child,
+    this.size = 40,
+    this.backgroundColor,
+  }) : super(key: key);
+
+  /// Main view that will be rendered at the center of the avatar.
+  /// It will feature a default icon size of `size / 2`.
+  final Widget child;
+
+  /// Size of the widget. Defaults to `40`.
+  final double size;
+
+  /// Main color that will be used to decorate the widget.
+  ///
+  /// If `null`, a random set of colors will be picked.
+  final AdwAvatarColors? backgroundColor;
+
+  @override
+  Widget build(BuildContext context) {
+    final colorPalette = AdwAvatarColorPalette.getColor(backgroundColor);
+    return SizedBox(
+      width: size,
+      height: size,
+      child: DecoratedBox(
```

Comment:
Why not use single Container instead of both SizedBox and DecoratedBox

Reply:
That's a pretty good question.

The `Container` widget has a ton of functionality built in, and it's generally advice to avoid its use because of the performance hit. It's much more economic in terms of resources to use to widget than using `Container` directly.

If you check its implementation, it's just adding those widgets to the widget tree, so it's cheaper to just use `SizedBox` and `DecoratedBox` directly.

Reply:
Thanks for the info!